oxide-auth/src/frontends/simple/endpoint.rs

```diff
@@ -114,6 +116,17 @@ pub struct Generic<R, A, I, S=Vacant, C=Vacant, L=Vacant> {
     pub response: L,
 }
 
+/// A simple wrapper around an Endpoint to change it's error type into anything `Into`-able.
+pub struct EndpointErrorMapper<E, Error>(E, PhantomData<Error>);
```

Comment:
Name bikeshedding: `MapErr`

It is terser and already in use for `futures`. `Map` is utilized in a number of places in the standard library as well and would also count as reasonably similar. The module path also contains `endpoint` already.

Reply:
Would it make sense, then, to add a `.map_err` method to the `Endpoint` trait?

Reply:
I've thought about adding it to a separate `EndpointExt` trait, yes. I just realized that this is not quite as generic as `MapErr` so maybe we shouldn't change this right now.

There can always be a typedef for this to refer to a more generic replacement, later.

Reply:
Actually, it might be more similar to `.from_err()`

Reply:
Remove the `Endpoint` portion of the name, however.

Reply:
`ErrorInto` would also be fine.